src/circuit.ts

```diff
@@ -49,12 +58,100 @@ export class Circuit {
+            32000: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final_15.ptau",
+            64000: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final_16.ptau",
+            128000: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final_17.ptau",
+            256000: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final_18.ptau",
+            512000: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final_19.ptau",
+            1000000: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final_20.ptau",
+            2000000: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final_21.ptau",
+            4000000: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final_22.ptau",
+            8000000: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final_23.ptau",
+            16000000: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final_24.ptau",
+            32000000: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final_25.ptau",
+            64000000: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final_26.ptau",
+            128000000: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final_27.ptau",
+            256000000: "https://hermez.s3-eu-west-1.amazonaws.com/powersOfTau28_hez_final.ptau",
+        }
+
+        for (var key in tauRangeMap) {
+            if (Object.prototype.hasOwnProperty.call(tauRangeMap, key)) {
+                if(c <= (parseInt(key))) {
+                    url = tauRangeMap[key]
+                    break;
+                }
+            }
+        }
+
+        return url
+    }
+ 
+    async downloadPowerOfTauFile(){
+        const tauFolderPath =  this._circuitConfig.outputDir + '/tau'
+
+        // tau folder will be created if it doesn't exist
+        if(!fs.existsSync(tauFolderPath)) {
+            fs.mkdirSync(tauFolderPath, {recursive:true})
+        }
+
+        // get tau file url according to total constraints
+        const totalConstraints = await this.getTotalConstraints()
+        log.info('TotalConstraints: %s', totalConstraints)
+        const tauFileUrl = this.getTauFileUrlByConstraints(totalConstraints)
+
+        // creating tau file name according to total constraints
+        let stringSplit = tauFileUrl.split("/")
+        let tauFileName = stringSplit[stringSplit.length - 1]
+        const localTaufilePath = path.resolve(tauFolderPath, tauFileName)
+
+        // change tau file path to the new file path
+        this._circuitConfig.powerOfTauFp = localTaufilePath
+       
+        // if the same tau file is already present, then return
+        if (fs.existsSync(tauFolderPath + "/" + tauFileName)) {
+            return
+        }
+
+        try{
+            await this.downloadFileOverHttp(tauFileUrl, localTaufilePath)
+        } catch(err){
+            console.log("Tau File Download Error: ", err)
```

Comment:
use log.error similar to log.info. Having log and console both is not a good idea. we should choose one and stick to it.